ingestion/src/metadata/ingestion/source/database/lineage_source.py

```diff
@@ -64,31 +66,38 @@ def get_table_query(self) -> Optional[Iterator[TableQuery]]:
                 f"Scanning query logs for {self.start.date()} - {self.end.date()}"
             )
             try:
-                with get_connection(self.service_connection).connect() as conn:
-                    rows = conn.execute(
-                        self.get_sql_statement(
-                            start_time=self.start,
-                            end_time=self.end,
-                        )
-                    )
-                    for row in rows:
-                        query_dict = dict(row)
-                        try:
-                            yield TableQuery(
-                                query=query_dict["query_text"],
-                                databaseName=self.get_database_name(query_dict),
-                                serviceName=self.config.serviceName,
-                                databaseSchema=self.get_schema_name(query_dict),
-                            )
-                        except Exception as exc:
-                            logger.debug(traceback.format_exc())
-                            logger.warning(
-                                f"Error processing query_dict {query_dict}: {exc}"
-                            )
+                engine = get_connection(self.service_connection)
+                yield from self.yield_table_query(engine)
+
             except Exception as exc:
                 logger.debug(traceback.format_exc())
                 logger.error(f"Source usage processing error: {exc}")
 
+    def yield_table_query(self, engine: Engine) -> Iterator[TableQuery]:
```

Comment:
Separating specific methods in lineage and usage source. This way we can directly call them in Vertica as we need to iterate over the engines (databases) to get all the data from different dbs

Reply:
Simplified this logic for Vertica. Although I am keeping the separation